dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -1115,6 +1115,12 @@ private TaskInstance findTaskIfExists(Long taskCode, int taskVersion) {
     private TaskInstance createTaskInstance(ProcessInstance processInstance, TaskNode taskNode) {
         TaskInstance taskInstance = findTaskIfExists(taskNode.getCode(), taskNode.getVersion());
         if (taskInstance != null) {
+            if (!taskInstance.getEnvironmentCode().equals(-1L)) {
+                Environment environment = processService.findEnvironmentByCode(taskInstance.getEnvironmentCode());
```

Comment:
I think we should not query again after workflow running, and I found that there was already in `WorkflowExecuteRunnable.newTaskInstance`.

Reply:
This bug will not appear in new tasks. This bug is that existing tasks will appear during fault tolerance.
` TaskInstance taskInstance = findTaskIfExists(taskNode.getCode(), taskNode.getVersion());`
findTaskIfExists return instance EnvironmentConfig is null at version 2.0.5 if new version findTaskIfExists method return TaskInstance EnvironmentConfig is not null no need to query again

Reply:
> This bug will not appear in new tasks. This bug is that existing tasks will appear during fault tolerance. ` TaskInstance taskInstance = findTaskIfExists(taskNode.getCode(), taskNode.getVersion());` findTaskIfExists return instance EnvironmentConfig is null at version 2.0.5 if new version findTaskIfExists method return TaskInstance EnvironmentConfig is not null no need to query again

In this case, we can add this logic into `initTaskQueue`, which will load the valid tasks at first if exist. What do you think?

Reply:
greate is a more elegant solution，i think in baseSql add environment_config column will resolve the problem
![image](https://user-images.githubusercontent.com/5456361/175458093-176da91c-9e29-4fc7-8335-aa4fe003483f.png)